### ISSUE TITLE
Lot of errors comparing 2 pdf files (from Excel files)

### ISSUE THREAD

**Issue by steevi2307** [red6/pdfcompare#86]
Hi,

We have errors when comparing two pdf files converted from Excel files
Do you have an idea of the problem ?

nov. 09, 2020 11:57:43 AM org.apache.pdfbox.pdmodel.font.FileSystemFontProvider loadDiskCache
AVERTISSEMENT: New fonts found, font cache will be re-built
nov. 09, 2020 11:57:43 AM org.apache.pdfbox.pdmodel.font.FileSystemFontProvider <init>
AVERTISSEMENT: Building on-disk font cache, this may take a while
nov. 09, 2020 11:57:43 AM org.apache.pdfbox.pdmodel.font.FileSystemFontProvider <init>
AVERTISSEMENT: Finished building on-disk font cache, found 373 fonts
11:57:46.736 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 7, x1: 1202, y1: 1277, x2: 1214, y2: 3240 }
11:57:46.927 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 8, x1: 1202, y1: 240, x2: 1214, y2: 3224 }
11:57:47.144 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 9, x1: 1202, y1: 240, x2: 1214, y2: 3226 }
11:57:47.473 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 10, x1: 1202, y1: 240, x2: 1214, y2: 3223 }
11:57:47.618 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 11, x1: 1202, y1: 240, x2: 1214, y2: 3223 }
11:57:47.768 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 12, x1: 1202, y1: 240, x2: 1214, y2: 2152 }
11:57:50.059 [Diff-3-thread-1] INFO de.redsix.pdfcompare.DiffImage - Differences found at { page: 24, x1: 450, y1: 2127, x2: 784, y2: 2152 }
11:58:04.124 [Draw-1-thread-1] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 102 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
        at java.awt.image.DataBufferInt.<init>(Unknown Source)
        at java.awt.image.Raster.createPackedRaster(Unknown Source)
        at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
        at java.awt.image.BufferedImage.<init>(Unknown Source)
        at org.apache.pdfbox.rendering.PDFRenderer.renderImage(PDFRenderer.java:296)
        at org.apache.pdfbox.rendering.PDFRenderer.renderImage(PDFRenderer.java:243)
        at org.apache.pdfbox.rendering.PDFRenderer.renderImageWithDPI(PDFRenderer.java:215)
        at de.redsix.pdfcompare.PdfComparator.renderPageAsImage(PdfComparator.java:529)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$10(PdfComparator.java:448)
        at de.redsix.pdfcompare.PdfComparator$$Lambda$7/1878150205.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        ... 3 common frames omitted
Exception in thread "Draw-1-thread-1" java.lang.OutOfMemoryError: Java heap space
        at java.awt.image.DataBufferInt.<init>(Unknown Source)
11:58:05.761 [Draw-1-thread-2] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 103 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
        at java.awt.image.Raster.createPackedRaster(Unknown Source)
        at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
11:58:07.078 [Draw-1-thread-3] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 104 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
        at java.awt.image.BufferedImage.<init>(Unknown Source)
        at de.redsix.pdfcompare.StacktraceImage.getImage(StacktraceImage.java:29)
        at de.redsix.pdfcompare.PdfComparator.addErrorPage(PdfComparator.java:477)11:58:08.160 [Draw-1-thread-4] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 105 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space

        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:466)
        at de.redsix.pdfcompare.PdfComparator$$Lambda$6/1778535015.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
11:58:09.664 [Draw-1-thread-5] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 106 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Exception in thread "Draw-1-thread-5" java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-4" java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-3" Exception in thread "Draw-1-thread-2" java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space11:58:11.688 [Draw-1-thread-6] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 107 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space

Exception in thread "Draw-1-thread-6" java.lang.OutOfMemoryError: Java heap space
11:58:12.823 [Draw-1-thread-7] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 108 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-7" java.lang.OutOfMemoryError: Java heap space
11:58:13.613 [Draw-1-thread-8] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 109 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-8" java.lang.OutOfMemoryError: Java heap space
11:58:14.449 [Draw-1-thread-9] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 110 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-9" java.lang.OutOfMemoryError: Java heap space
11:58:15.250 [Draw-1-thread-10] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 111 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-10" java.lang.OutOfMemoryError: Java heap space
11:58:16.126 [Draw-1-thread-11] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 112 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-11" java.lang.OutOfMemoryError: Java heap space
11:58:16.914 [Draw-1-thread-12] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 113 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-12" java.lang.OutOfMemoryError: Java heap space
11:58:17.693 [Draw-1-thread-13] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 114 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-13" java.lang.OutOfMemoryError: Java heap space
11:58:18.614 [Draw-1-thread-14] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 115 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-14" java.lang.OutOfMemoryError: Java heap space
11:58:19.446 [Draw-1-thread-15] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 116 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-15" java.lang.OutOfMemoryError: Java heap space
11:58:20.576 [Draw-1-thread-16] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 117 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-16" java.lang.OutOfMemoryError: Java heap space
11:58:21.401 [Draw-1-thread-17] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 118 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-17" java.lang.OutOfMemoryError: Java heap space
11:58:22.235 [Draw-1-thread-18] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 119 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-18" java.lang.OutOfMemoryError: Java heap space
11:58:22.820 [Draw-1-thread-19] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 120 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-19" java.lang.OutOfMemoryError: Java heap space
11:58:23.882 [Draw-1-thread-20] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 121 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-20" java.lang.OutOfMemoryError: Java heap space
11:58:24.681 [Draw-1-thread-21] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 122 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-21" java.lang.OutOfMemoryError: Java heap space
11:58:25.457 [Draw-1-thread-22] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 123 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-22" java.lang.OutOfMemoryError: Java heap space
11:58:26.232 [Draw-1-thread-23] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 124 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-23" java.lang.OutOfMemoryError: Java heap space
11:58:27.025 [Draw-1-thread-24] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 125 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-24" java.lang.OutOfMemoryError: Java heap space
11:58:27.814 [Draw-1-thread-25] ERROR de.redsix.pdfcompare.PdfComparator - An error occurred, while rendering this page
de.redsix.pdfcompare.RenderingException: Error while rendering page 126 of expected document
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
        at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
        ... 4 common frames omitted
Caused by: java.lang.OutOfMemoryError: Java heap space
Exception in thread "Draw-1-thread-25" java.lang.OutOfMemoryError: Java heap space
Exception in thread "main" de.redsix.pdfcompare.RenderingException: Exceptions where caught during rendering or diffing
        at de.redsix.pdfcompare.PdfComparator.compare(PdfComparator.java:435)
        at de.redsix.pdfcompare.PdfComparator.compare(PdfComparator.java:391)
        at Test1.main(Test1.java:10)
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 102 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        Caused by: java.lang.OutOfMemoryError: Java heap space
                at java.awt.image.DataBufferInt.<init>(Unknown Source)
                at java.awt.image.Raster.createPackedRaster(Unknown Source)
                at java.awt.image.DirectColorModel.createCompatibleWritableRaster(Unknown Source)
                at java.awt.image.BufferedImage.<init>(Unknown Source)
                at org.apache.pdfbox.rendering.PDFRenderer.renderImage(PDFRenderer.java:296)
                at org.apache.pdfbox.rendering.PDFRenderer.renderImage(PDFRenderer.java:243)
                at org.apache.pdfbox.rendering.PDFRenderer.renderImageWithDPI(PDFRenderer.java:215)
                at de.redsix.pdfcompare.PdfComparator.renderPageAsImage(PdfComparator.java:529)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$10(PdfComparator.java:448)
                at de.redsix.pdfcompare.PdfComparator$$Lambda$7/1878150205.call(Unknown Source)
                at java.util.concurrent.FutureTask.run(Unknown Source)
                ... 3 more
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 103 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        Caused by: java.lang.OutOfMemoryError: Java heap space
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 104 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 105 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 106 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 107 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 108 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 109 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 110 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 111 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 112 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 113 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 114 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 115 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 116 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 117 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 118 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 119 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 120 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 121 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 122 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 123 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 124 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 125 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]
        Suppressed: de.redsix.pdfcompare.RenderingException: Error while rendering page 126 of expected document
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:491)
                at de.redsix.pdfcompare.PdfComparator.lambda$drawImage$13(PdfComparator.java:451)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
                at java.lang.Thread.run(Unknown Source)
        Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
                at java.util.concurrent.FutureTask.report(Unknown Source)
                at java.util.concurrent.FutureTask.get(Unknown Source)
                at de.redsix.pdfcompare.PdfComparator.getImage(PdfComparator.java:483)
                ... 4 more
        [CIRCULAR REFERENCE:java.lang.OutOfMemoryError: Java heap space]

Thanks ;)

**Comment by gastendonk** [comment]
I suggest to try this option:

        SimpleEnvironment env = new SimpleEnvironment();
        env.setAddEqualPagesToResult(false);

And why do you verify Excel files using PDF? You can compare Excel files using Apache POI.

**Comment by finsterwalder** [comment]
Comparing Excel files as a PDF does sound strange...

But back to the problem: You are simply running out of memory. PdfCompare needs quite a lot of memory per page compared.You can reduce the amount of memory used by not storing equal pages, as @gastendonk suggested.
But to completely solves this problem, look at the compare results with overflow:
https://github.com/red6/pdfcompare#different-compareresult-implementations

**Comment by steevi2307** [comment]
Hi,
I'm running 16go of RAM and during comparison, the memory goes up to 10go (java process goes up to 4go)
So do you think it's really an out of memory error ?

**Comment by finsterwalder** [comment]
Your JVM definitely runs out of memory. That what it tells you here:

Caused by: java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space

**Comment by steevi2307** [comment]
Hi,

And do you know we can bypass this ?
Because we have two other files to compare (8mo pdf - 670 pages) and the I have now the same problem ...

Memory usage : 57% (java +/- 4.5Go)

https://drive.google.com/file/d/10kM0w7bOLd5juEJOFZU8xhFPi50jrbRj/view?usp=sharing
https://drive.google.com/file/d/1ytTnUDDSIsg3vSeTe7pe4hGXYbhzD6PV/view?usp=sharing

Thanks ;)

**Comment by finsterwalder** [comment]
Like I said above:

> But to completely solves this problem, look at the compare results with overflow:
https://github.com/red6/pdfcompare#different-compareresult-implementations

**Comment by steevi2307** [comment]
Hi,
Do you have a sample to simply to limit the memory in our case :

final CompareResult result = new PdfComparator(x, y).withEnvironment(new SimpleEnvironment().setAddEqualPagesToResult(false)).compare();

I don't have to use CompareResult anymore if I understand ?

Thanks

**Comment by finsterwalder** [comment]
Just use
`new PdfComparator(x, y, new CompareResultWithPageOverflow())`
or
`new PdfComparator(x, y, new CompareResultWithPageOverflow(<maxNrOfPages>))`

**Comment by steevi2307** [comment]
And it's possible in this way to have a "diff" file showing differences as previous call ?

**Comment by finsterwalder** [comment]
The result is the same. The only difference is, that compared pages are swapped to disk and merged at the end.

**Comment by steevi2307** [comment]
It seems to work ; thanks ! :)

**Comment by steevi2307** [comment]
Hi,

Another question : everything works fine but the comparison is a little bit long for large pdf files (50 minutes for 1500 pages)
Is there a way to optimise this ?

Thanks

**Comment by steevi2307** [comment]
Hi,

Another question : everything works fine but the comparison is a little bit long for large pdf files (50 minutes for 1500 pages)
Is there a way to optimise this ?

Thanks

**Comment by finsterwalder** [comment]
General note: Please don't reopen tickets to discuss a completely different subject. Open a new issue instead.

Unfortunately there is nothing easy you can do to improve performance. I've been playing around with more parallel threads, but the underlying PdfBox library I use is not easily thread safe when rendering pages. Rendering pages and comparing them is already running in parallel.
